src/node.ts

```diff
@@ -393,46 +395,75 @@ export abstract class BaseNode<
    * @returns a mapping of name to the node visible in this node's scope.
    */
   public getLexicalScope(): Map<string, BindingDecl> {
-    return new Map(getLexicalScope(this as unknown as FunctionlessNode));
+    return new Map(
+      getLexicalScope(this as unknown as FunctionlessNode, "scope")
+    );
 
-    function getLexicalScope(node: FunctionlessNode | undefined): Binding[] {
+    /**
+     * @param kind the relation of the current `node` to the requesting `node`.
+     *             * `scope` - the current node is an ancestor of the requesting node
+     *             * `sibling` - the current node is the sibling of the requesting node
+     *             ```ts
+     *             for(const i in []) { // scope - emits i=self
+     *                const a = ""; // sibling - emits a=self
+     *                for(const a of []) {} // sibling emits []
+     *                a // requesting node
+     *             }
+     *             ```
+     *             some nodes only emit names to their `scope` (ex: for) and
+     *             other nodes emit names to all of their `sibling`s (ex: variableStmt)
+     */
```

Comment:
Strange indentation

Reply:
hmm, I lined it up with the `the` after `kind`, but it works to line it up with the `kind` instead.

Reply:
![image](https://user-images.githubusercontent.com/289213/183470502-b5b9a45a-47ef-4489-8a3b-b416bb2bb9a1.png)

Reply:
Maybe move the doc to the actual parameter declaration? That indentation hurts